src/JSON/Deserializer.dfy

```diff
@@ -0,0 +1,151 @@
+/// ===============================================
+///  Deserialization from JSON.Grammar to JSON.AST
+/// ===============================================
+///
+/// For the spec, see ``JSON.Spec.dfy``.
+
+include "../Collections/Sequences/Seq.dfy"
+include "../BoundedInts.dfy"
+
+include "Utils/Views.dfy"
+include "Utils/Vectors.dfy"
+include "Utils/Unicode.dfy"
+include "Errors.dfy"
+include "AST.dfy"
+include "Grammar.dfy"
+include "Spec.dfy"
+
+module {:options "-functionSyntax:4"} JSON.Deserializer {
+  import Seq
+
+  import opened Wrappers
+  import opened BoundedInts
+  import opened Logarithm
+  import opened Power
+  import opened Utils.Str
+  import Utils.Unicode
+
+  import AST
+  import Spec
+  import opened Errors
+  import opened Utils.Vectors
+  import opened Grammar
+  import opened Utils.Views.Core
+
+  function Bool(js: Grammar.jbool): bool {
+    assert js.Bytes() in {Grammar.TRUE, Grammar.FALSE};
+    js.At(0) == 't' as byte
+  }
+
+  function Unescape(str: string, start: nat := 0): DeserializationResult<string>
```

Comment:
If you did not write a verifier for Escape/Unescape, could you please add a comment in front of each function?

Reply:
Done, but just in front of Unescape: `Escape` is the spec :)